engine/src/main/java/org/camunda/bpm/engine/impl/cfg/auth/DefaultAuthorizationProvider.java

```diff
@@ -292,10 +294,53 @@
           updateAuthorization(historyAuthorization, userId, groupId, HISTORIC_TASK,
               taskId, HistoricTaskPermissions.READ);
 
+      String rootProcessInstanceId = getRootProcessInstanceId(task);
+
+      historyAuthorization.setRootProcessInstanceId(rootProcessInstanceId);
+
+      if (isHistoryRemovalTimeStrategyStart()) {
```

Comment:
In case of standalone tasks (root instance id == null), we don't have to go looking for the root instance. 

Which removal time should a standalone task permissions receive?

Reply:
👍